add field for billboard target geolocations, with unit and e2e tests

dukegreene opened the pull request:
<!--
     For Work In Progress Pull Requests, please use the Draft PR feature,
     see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.

     For a timely review/response, please avoid force-pushing additional
     commits if your PR already received reviews or comments.

     Before submitting a Pull Request, please ensure you've done the following:
     - 📖 Read the Forem Contributing Guide: https://developers.forem.com/contributing-guide/forem#create-a-pull-request
     - 📖 Read the Forem Code of Conduct: https://github.com/forem/forem/blob/main/CODE_OF_CONDUCT.md
     - 👷‍♀️ Create small PRs. In most cases this will be possible.
     - ✅ Provide tests for your changes.
     - 📝 Use descriptive commit messages.
     - 📗 Update any related documentation and include any relevant screenshots.

     NOTE: Pull Requests from forked repositories will need to be reviewed by
     a Forem Team member before any CI builds will run. Once your PR is approved
     with a `/ci` reply to the PR, it will be allowed to run subsequent builds without
     manual approval.
-->

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Following up on https://github.com/forem/forem/pull/19834, this change adds a field during the creation and updating of a Billboard, and lets users set comma-separated ISO3166-2 geolocation codes (US and Canada only) according to the validations already present on the model.

This change didn't have to be too large because the model is already good at displaying specific error messages for its failed validations. We're just adding a small helper method to ensure the typed input persists in its comma-separated format as the user resubmits and the page refreshes.

## Related Tickets & Documents

<!--
For pull requests that relate or close an issue, please include them
below.  We like to follow [Github's guidance on linking issues to pull requests](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

For example having the text: "closes #1234" would connect the current pull
request to issue 1234.  And when we merge the pull request, Github will
automatically close the issue.
-->

- Related Issue #19794
- Closes #19794

## QA Instructions, Screenshots, Recordings

The feature flag `billboard_location_targeting` needs to be enabled to test this functionality. In localhost, I did this by going to http://localhost:3000/admin/feature_flags/features and turning it all the way on, and I believe the Admin test account will be able to do a similar toggling in Uffizzi once I draft this PR.

With the flag enabled, whether creating or editing a billboard ("Admin -> Customization -> Display Ads"), you should now see a "Target Geolocations: field with placeholder examples of the accepted format and the countries we currently support.

Errors will generate from that field if you 
  attempt to create/update with an unsupported country's valid code e.g. "MX-CMX" or
  attempt to create/update with a supported country and invalid region e.g. "US-UHHHH"
  
With otherwise valid inputs, a string of comma separated valid inputs for geolocations e.g. "US-NV, US-TX, CA-BC, CA-NS" should allow a successful save and persist on that display ad when its Show page is loaded.

If some inputs are valid and others are invalid, only the invalid ones should show up in the flash notice about form errors.

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [X] Semantic HTML implemented?
- [X] Keyboard operability supported? Unrelated note to self: tabbing to the "Display to user group" and "Type" radio buttons works, but there's no visual indicator that the radio groups are currently tabbed to like there is with outlined text fields. Will see if there's already an issue for this.
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [ ] Color contrast tested?

_For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [X] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [optional] Are there any post deployment tasks we need to perform?

Make sure the flipper feature is enabled for the actors who plan to test it

## [optional] What gif best describes this PR or how it makes you feel?

![Gina Rodriguez from the Netflix Carmen Sandiego saying "I wanted you to find me"](https://media.tenor.com/JSxQlfeP5WsAAAAC/i-wanted-you-to-find-me-gina-rodriguez.gif)

filleduchaos commented on app/models/display_ad.rb:
```diff
@@ -173,6 +173,10 @@ def exclude_article_ids=(input)
   end
 
+  def comma_separated_geolocations
+    target_geolocations.map { |g| "#{g.to_iso3166}, " }.join.chomp(", ")
```
I _think_ you can accomplish same with `.map(&:to_iso3166).join(", ")`?

lightalloy replied:
It seems like this code belongs more to a helper than to a model :eyes: 
(as it's used to format data in the view)

filleduchaos reviewed (approved):
QA'd locally and works great 🎉

lightalloy reviewed (commented):
Works well.
I suggest implementing changes that @filleduchaos and me mentioned.
I would also add a couple of tests to `spec/requests/admin/billboards_spec.rb` (though it's kind of covered with e2e tests)

There is also a merge conflict because renaming display ads to billboards is in process, so need to update and resolve.

dukegreene commented:
Thanks @lightalloy and @filleduchaos for the helpful suggestions! I moved the refactored method work inline and got rid of the spec for the method, and also added some request specs for context

lightalloy reviewed (approved):
Thanks for implementing the changes :raised_hands:
